packages/web-embed-api/src/typeGuards.ts

```diff
@@ -0,0 +1,58 @@
+import { WebEmbedMessage } from '@formsort/constants';
+import {
+  IIFrameAnalyticsEventData,
+  IIFrameRedirectEventData,
+  IIFrameResizeEventData,
+  IWebEmbedEventData,
+} from './interfaces';
+
+function isRecord(val: unknown): val is { [key: string]: unknown } {
+  return val !== null && typeof val === 'object';
+}
+
+function isString(val: unknown): val is string {
+  return typeof val === 'string';
+}
+
```

Comment:
I wouldn't create this function as it just checks `string` type

Reply:
Yeah I agree that how it's used now this is a bit simple to put in a function. I do have a plan for it for the future, where we can have type guards that check for interfaces. For example, having an `isShape` function:

```typescript
interface Student {
   name: string,
   age: number
}

const isStudent = isShape<Student>({ 
   name: isString,
   age: isNumber
})

function main(val: unknown) {
   if (isStudent(val)){
      // val is Student
   }
}
```
There are some existing libraries that use a similar approach (e.g. [zod](https://github.com/colinhacks/zod) and [Runtypes](https://github.com/pelotom/runtypes))